src/Clay/Size.hs

```diff
@@ -88,12 +91,19 @@ data LengthUnit
 data Percentage
 
 -- | When combining percentages with units using calc, we get a combination
-data Combination
+-- | Any unit or combination of units
+data AnyUnit
+
+-- | Upcast a size unit
+upcast :: Size a -> Size AnyUnit
+upcast = coerce
```

Comment:
Can you elaborate?
What's the functional difference from the user perspective?
It's more work to embed it in a constructor, I now have to add cases to `sizeToText` & the `Val` instance that do nothing but unwrap.

Reply:
From the user perspective it's mostly the same I hope. But from the maintainer's perspective a huge burden is gone because noone has to worry about `coerce` breaking anything.

Reply:
I have to admit that I completely misunderstood what you did here. I mistook `coerce` for `unsafeCoerce`, which is - as the name says - unsafe, whereas `coerce` is completely fine. It will never break anything because the type checker ensures that coercion only happens when the internal representations are indeed the same. If that was your goal, I'm sorry about the misdirection: feel free to disregard the stuff I said about GADTs, and let's make everyone's life easier by using `coerce`.